File: cat.php
#!/usr/bin/env php
<?php 
    if ($argc > 1) {
        if (file_exists($argv[1])) {
            readfile($argv[1], false, NULL);
            echo("\n");
        } else {
            exit(print("File not found\n")); //This uses the property of print that it always returns 1
        }
    } else {
        exit(print("File not found\n")); 
    }
?>

File: cat.dart
import "dart:io";

void main(List<String> argv) {
    if (argv.length > 0) {
        final file = File(argv[0]);
        file.readAsString().then((value) => print(value));
    } else {
        print("File not found");
        exit(1);
    } 
}